File: app/src/components/TakimFiksturModal.js
// TAKIM FİKSTÜRÜ — oynadığı ve oynayacağı maçlar.
// ---------------------------------------------------------------------------
// Maç detayında takım adının altındaki bağlantıdan açılır. Eskiden burada
// sezon istatistikleri vardı; kullanıcı kararıyla fikstüre çevrildi.
//
// TASARIM KARARI — tek liste, tarih sırası, "bugün" çizgisi:
// Oynanmış ve oynanacak maçlar AYRI sekmelere bölünmedi. Bir takımın gidişatı
// okunurken en çok işe yarayan şey, biten maçların hemen ardından sıradaki
// maçı görmek. Ayırmak, kullanıcıyı iki liste arasında gidip gelmeye zorlardı.
// Bunun yerine oynanmışlar ile oynanacaklar arasına "SIRADAKİ" ayracı konur ve
// liste ilk açılışta oraya kaydırılır.
//
// SONUÇ HARFİ YALNIZ BİTMİŞ MAÇTA: G/B/M rozeti oynanmamış maça basılmaz,
// skor yoksa yazılmaz. Sunucu da aynı kuralı uygular (takimFikstur.js).
import React, { useEffect, useState, useMemo, useRef } from 'react';
import {
  Modal, View, Text, ScrollView, TouchableOpacity, ActivityIndicator, StyleSheet,
} from 'react-native';

import { api } from '../api';
import { colors, spacing, radius } from '../theme';
import { Logo } from '../ui';

const AY = ['Oca', 'Şub', 'Mar', 'Nis', 'May', 'Haz', 'Tem', 'Ağu', 'Eyl', 'Eki', 'Kas', 'Ara'];

export function tarihEtiketi(dateUnix) {
  if (!dateUnix) return '—';
  const d = new Date(dateUnix * 1000);
  if (Number.isNaN(d.getTime())) return '—';
  return `${d.getDate()} ${AY[d.getMonth()]}`;
}

const SONUC_RENGI = { G: colors.green, B: colors.yellow, M: colors.red };

function SatirMac({ f, ligYaz }) {
  const evde = !!f.evde;
  const bizKazandi = f.sonuc;
  return (
    <View style={[st.satir, !f.oynandi && st.satirGelecek]}>
      <View style={st.solSutun}>
        <Text style={st.tarih} numberOfLines={1}>{tarihEtiketi(f.dateUnix)}</Text>
        {/* Turnuva adı YALNIZ birden fazla turnuva varsa yazılır — tek
            turnuvada her satıra aynı adı basmak gürültüden ibaret olurdu. */}
        {ligYaz && f.lig ? <Text style={st.ligTxt} numberOfLines={1}>{f.lig}</Text> : null}
      </View>

      <View style={st.orta}>
        <Text style={[st.takim, evde && st.takimBiz]} numberOfLines={1}>{f.home || '—'}</Text>
        {/* Skor YALNIZ oynanmış maçta; oynanmamışta nötr ayraç. */}
        {f.oynandi && f.score ? (
          <Text style={st.skor}>{f.score.home} - {f.score.away}</Text>
        ) : (
          <Text style={st.skorYok}>v</Text>
        )}
        <Text style={[st.takim, st.takimSag, !evde && st.takimBiz]} numberOfLines={1}>{f.away || '—'}</Text>
      </View>

      {bizKazandi ? (
        <View style={[st.rozet, { backgroundColor: SONUC_RENGI[bizKazandi] || colors.gray }]}>
          <Text style={st.rozetTxt}>{bizKazandi}</Text>
        </View>
      ) : (
        <View style={st.rozetBos} />
      )}
    </View>
  );
}

export default function TakimFiksturModal({ visible, onClose, takim }) {
  const { teamId, seasonId, name, logo, league } = takim || {};
  const [veri, setVeri] = useState(null);
  const [hata, setHata] = useState(null);
  const [yukleniyor, setYukleniyor] = useState(false);
  const kaydirmaRef = useRef(null);
  const ayracYRef = useRef(0);

  useEffect(() => {
    if (!visible || !teamId || !seasonId) return undefined;
    let canli = true;
    setVeri(null); setHata(null); setYukleniyor(true);
    api.teamFixtures(teamId, seasonId)
      .then((r) => { if (canli) setVeri(r); })
      .catch((e) => { if (canli) setHata(e.message); })
      .finally(() => { if (canli) setYukleniyor(false); });
    return () => { canli = false; };
  }, [visible, teamId, seasonId]);

  const fikstur = veri?.fikstur || [];
  // Ayraç konumu: ilk oynanmamış maç. Hiç yoksa (sezon bitti) ayraç çizilmez.
  const ilkGelecek = useMemo(() => fikstur.findIndex((f) => !f.oynandi), [fikstur]);
  const ligler = useMemo(
    () => [...new Set(fikstur.map((f) => f.lig).filter(Boolean))],
    [fikstur],
  );

  return (
    <Modal visible={!!visible} transparent animationType="slide" onRequestClose={onClose}>
      <View style={st.overlay}>
        <TouchableOpacity style={st.backdrop} activeOpacity={1} onPress={onClose} />
        <View style={st.sheet}>
          <View style={st.head}>
            <Logo uri={logo} name={name} size={40} />
            <View style={{ flex: 1 }}>
              <Text style={st.ad} numberOfLines={1}>{veri?.takimAdi || name || '—'}</Text>
              <Text style={st.alt} numberOfLines={1}>
                {league || ''}
                {veri ? ` · ${veri.oynanan}/${veri.toplam} maç oynandı` : ''}
              </Text>
            </View>
            <TouchableOpacity onPress={onClose} hitSlop={10} accessibilityRole="button" accessibilityLabel="Kapat"><Text style={st.kapat}>✕</Text></TouchableOpacity>
          </View>

          {yukleniyor ? (
            <View style={st.merkez}><ActivityIndicator color={colors.primary} /></View>
          ) : hata ? (
            // Hata GİZLENMEZ. "Maçı yok" ile "veri alınamadı" farklı şeyler.
            <View style={st.merkez}>
              <Text style={st.hataBaslik}>Fikstür alınamadı</Text>
              <Text style={st.hataTxt}>{hata}</Text>
            </View>
          ) : !fikstur.length ? (
            <View style={st.merkez}>
              <Text style={st.hataTxt}>Bu takım için sezon maçı bulunamadı.</Text>
            </View>
          ) : (
            <ScrollView
              ref={kaydirmaRef}
              contentContainerStyle={{ paddingBottom: spacing.xl }}
              showsVerticalScrollIndicator={false}
              onContentSizeChange={() => {
                // Sıradaki maçı görünür yap: kullanıcı fikstürü açtığında
                // sezonun başındaki eski maçlara değil, buraya bakmak ister.
                if (ayracYRef.current > 0) {
                  kaydirmaRef.current?.scrollTo({ y: Math.max(0, ayracYRef.current - 60), animated: false });
                }
              }}
            >
              {fikstur.map((f, i) => (
                <View
                  key={`${f.footyMatchId ?? i}`}
                  onLayout={i === ilkGelecek ? (e) => { ayracYRef.current = e.nativeEvent.layout.y; } : undefined}
                >
                  {i === ilkGelecek ? (
                    <View style={st.ayrac}>
                      <View style={st.ayracCizgi} />
                      <Text style={st.ayracTxt}>SIRADAKİ</Text>
                      <View style={st.ayracCizgi} />
                    </View>
                  ) : null}
                  <SatirMac f={f} ligYaz={ligler.length > 1} />
                </View>
              ))}

              {/* KAPSAM NOTU — hangi turnuvaların dahil olduğu YAZILIR.
                  Liste dolu göründüğü için, kupa/Avrupa maçlarının eksik
                  olduğu ancak burada anlaşılır. Sessiz eksik, yanlış bilgiden
                  daha tehlikelidir. */}
              {ligler.length ? (
                <Text style={st.kapsam}>
                  Kapsam: {ligler.join(' · ')}
                </Text>
              ) : null}
            </ScrollView>
          )}
        </View>
      </View>
    </Modal>
  );
}

const st = StyleSheet.create({
  overlay: { flex: 1, justifyContent: 'flex-end' },
  backdrop: { ...StyleSheet.absoluteFillObject, backgroundColor: 'rgba(0,0,0,0.45)' },
  sheet: {
    maxHeight: '86%',
    backgroundColor: colors.card,
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
    overflow: 'hidden',
  },
  head: {
    flexDirection: 'row', alignItems: 'center', gap: spacing.sm,
    padding: spacing.lg, borderBottomWidth: 1, borderBottomColor: colors.border,
  },
  ad: { color: colors.text, fontSize: 16, fontWeight: '900' },
  alt: { color: colors.textMuted, fontSize: 11.5, fontWeight: '700', marginTop: 2 },
  kapat: { color: colors.textMuted, fontSize: 18, fontWeight: '900', paddingHorizontal: 4 },

  merkez: { padding: spacing.xl, alignItems: 'center', gap: 6 },
  hataBaslik: { color: colors.text, fontSize: 14, fontWeight: '900' },
  hataTxt: { color: colors.textMuted, fontSize: 12.5, textAlign: 'center' },

  satir: {
    flexDirection: 'row', alignItems: 'center',
    paddingVertical: 9, paddingHorizontal: spacing.lg,
    borderBottomWidth: 1, borderBottomColor: colors.border,
  },
  satirGelecek: { backgroundColor: colors.bgAlt },
  solSutun: { width: 76 },
  tarih: { color: colors.textMuted, fontSize: 11, fontWeight: '800' },
  ligTxt: { color: colors.textMuted, fontSize: 9, fontWeight: '700', marginTop: 1, opacity: 0.8 },
  kapsam: {
    color: colors.textMuted, fontSize: 10.5, fontWeight: '700',
    paddingHorizontal: spacing.lg, paddingTop: spacing.md, textAlign: 'center',
  },
  orta: { flex: 1, flexDirection: 'row', alignItems: 'center' },
  takim: { flex: 1, color: colors.text, fontSize: 12.5, fontWeight: '600' },
  takimSag: { textAlign: 'right' },
  takimBiz: { fontWeight: '900' },        // kartı açılan takım kalın
  skor: { width: 46, textAlign: 'center', color: colors.text, fontSize: 12.5, fontWeight: '900' },
  skorYok: { width: 46, textAlign: 'center', color: colors.textMuted, fontSize: 11, fontWeight: '700' },

  rozet: { width: 20, height: 20, borderRadius: 10, alignItems: 'center', justifyContent: 'center', marginLeft: 8 },
  rozetTxt: { color: '#fff', fontSize: 10, fontWeight: '900' },
  rozetBos: { width: 20, marginLeft: 8 },

  ayrac: { flexDirection: 'row', alignItems: 'center', gap: 8, paddingHorizontal: spacing.lg, paddingVertical: 8 },
  ayracCizgi: { flex: 1, height: 1, backgroundColor: colors.border },
  ayracTxt: { color: colors.primary, fontSize: 10, fontWeight: '900', letterSpacing: 0.5 },
});

File: flutter/lib/features/match_detail/takim_fikstur_modal.dart
// KAYNAK: app/src/components/TakimFiksturModal.js — BİREBİR çeviri.
//
// TAKIM FİKSTÜRÜ — oynadığı ve oynayacağı maçlar.
// Maç detayında takım adının altındaki bağlantıdan açılır. Eskiden burada
// sezon istatistikleri vardı; kullanıcı kararıyla fikstüre çevrildi.
//
// TASARIM KARARI — tek liste, tarih sırası, "sıradaki" çizgisi:
// Oynanmış ve oynanacak maçlar AYRI sekmelere bölünmedi. Bir takımın gidişatı
// okunurken en çok işe yarayan şey, biten maçların hemen ardından sıradaki
// maçı görmek. Ayırmak, kullanıcıyı iki liste arasında gidip gelmeye zorlardı.
// Bunun yerine oynanmışlar ile oynanacaklar arasına "SIRADAKİ" ayracı konur ve
// liste ilk açılışta oraya kaydırılır.
//
// SONUÇ HARFİ YALNIZ BİTMİŞ MAÇTA: G/B/M rozeti oynanmamış maça basılmaz,
// skor yoksa yazılmaz. Sunucu da aynı kuralı uygular (takimFikstur.js).

import 'dart:math' as math;

import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

import '../../core/network/api_client.dart';
import '../../core/takim_fikstur.dart';
import '../../core/theme/takim_paleti.dart' show okunurMetin;
import '../../core/theme/tokens.dart';
import '../../widgets/app_ui.dart';

const Map<String, Color> _sonucRengi = {
  'G': AppColors.green,
  'B': AppColors.yellow,
  'M': AppColors.red,
};

/// Modalı açan tek giriş noktası. Kaynakta `visible` bayrağıyla çizilen
/// `<Modal animationType="slide">` karşılığı: alttan kayan sayfa.
Future<void> takimFiksturuAc(
  BuildContext context, {
  required Object? teamId,
  required Object? seasonId,
  String? name,
  String? logo,
  String? league,
}) {
  return showModalBottomSheet<void>(
    context: context,
    isScrollControlled: true,
    backgroundColor: Colors.transparent,
    // Kaynaktaki yarı saydam siyah zemin (rgba(0,0,0,0.45)).
    barrierColor: const Color(0x73000000),
    builder: (_) => TakimFiksturSayfasi(
      teamId: teamId,
      seasonId: seasonId,
      name: name,
      logo: logo,
      league: league,
    ),
  );
}

class TakimFiksturSayfasi extends StatefulWidget {
  const TakimFiksturSayfasi({
    super.key,
    this.teamId,
    this.seasonId,
    this.name,
    this.logo,
    this.league,
  });

  final Object? teamId;
  final Object? seasonId;
  final String? name;
  final String? logo;
  final String? league;

  @override
  State<TakimFiksturSayfasi> createState() => _TakimFiksturSayfasiState();
}

class _TakimFiksturSayfasiState extends State<TakimFiksturSayfasi> {
  Map? _veri;
  String? _hata;
  bool _yukleniyor = false;

  final ScrollController _kaydirma = ScrollController();
  final GlobalKey _ayracKey = GlobalKey();
  bool _kaydirildi = false;

  @override
  void initState() {
    super.initState();
    // Kimlik yoksa istek HİÇ atılmaz — kaynakta da `if (!teamId || !seasonId)`.
    if (widget.teamId == null || widget.seasonId == null) return;
    _yukleniyor = true;
    _getir();
  }

  Future<void> _getir() async {
    try {
      final r = await api.teamFixtures(widget.teamId!, widget.seasonId!);
      if (!mounted) return;
      setState(() {
        _veri = r as Map?;
        _yukleniyor = false;
      });
    } catch (e) {
      if (!mounted) return;
      setState(() {
        // Kaynakta `e.message` basılıyordu; API katmanı hatayı zaten
        // kullanıcıya okunur metinle fırlatıyor.
        _hata = '$e';
        _yukleniyor = false;
      });
    }
  }

  @override
  void dispose() {
    _kaydirma.dispose();
    super.dispose();
  }

  /// Sıradaki maçı görünür yapar: kullanıcı fikstürü açtığında sezonun
  /// başındaki eski maçlara değil, buraya bakmak ister.
  ///
  /// Kaynakta `onContentSizeChange` içinde `scrollTo({ y: ayracY - 60 })`
  /// vardı; burada aynı hesap ilk kare çizildikten sonra yapılır.
  void _ayraciGoster() {
    if (_kaydirildi) return;
    final ctx = _ayracKey.currentContext;
    if (ctx == null) return;
    final box = ctx.findRenderObject();
    if (box is! RenderBox || !box.hasSize) return;
    if (!_kaydirma.hasClients) return;
    final y = RenderAbstractViewport.of(box).getOffsetToReveal(box, 0).offset;
    _kaydirildi = true;
    _kaydirma.jumpTo(
      math.max(0.0, y - 60).clamp(0.0, _kaydirma.position.maxScrollExtent),
    );
  }

  @override
  Widget build(BuildContext context) {
    final fikstur = (_veri?['fikstur'] as List?) ?? const [];
    final ilkGelecek = ilkGelecekIndex(fikstur);
    final ligler = fiksturLigleri(fikstur);

    return Padding(
      // Kaynakta `justifyContent: 'flex-end'` — sayfa ekranın altına yaslanır.
      padding: EdgeInsets.only(top: MediaQuery.of(context).padding.top),
      child: Align(
        alignment: Alignment.bottomCenter,
        child: ConstrainedBox(
          constraints: BoxConstraints(
            maxHeight: MediaQuery.of(context).size.height * 0.86,
          ),
          child: Container(
            decoration: BoxDecoration(
              color: AppColors.card,
              borderRadius: BorderRadius.only(
                topLeft: Radius.circular(20),
                topRight: Radius.circular(20),
              ),
            ),
            clipBehavior: Clip.antiAlias,
            child: Column(
              mainAxisSize: MainAxisSize.min,
              children: [
                _baslik(),
                Flexible(child: _icerik(fikstur, ilkGelecek, ligler)),
              ],
            ),
          ),
        ),
      ),
    );
  }

  Widget _baslik() {
    final oynanan = _veri?['oynanan'];
    final toplam = _veri?['toplam'];
    return Container(
      padding: const EdgeInsets.all(Spacing.lg),
      decoration: BoxDecoration(
        border: Border(bottom: BorderSide(color: AppColors.border)),
      ),
      child: Row(
        children: [
          Logo(uri: widget.logo, name: widget.name, size: 40),
          const SizedBox(width: Spacing.sm),
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  '${_veri?['takimAdi'] ?? widget.name ?? '—'}',
                  maxLines: 1,
                  overflow: TextOverflow.ellipsis,
                  style: TextStyle(
                    color: AppColors.text,
                    fontSize: 16,
                    fontWeight: AppFont.black,
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.only(top: 2),
                  child: Text(
                    '${widget.league ?? ''}'
                    '${_veri != null ? ' · $oynanan/$toplam maç oynandı' : ''}',
                    maxLines: 1,
                    overflow: TextOverflow.ellipsis,
                    style: TextStyle(
                      color: AppColors.textMuted,
                      fontSize: 11.5,
                      fontWeight: AppFont.bold,
                    ),
                  ),
                ),
              ],
            ),
          ),
          const SizedBox(width: Spacing.sm),
          Semantics(
            button: true,
            label: 'Kapat',
            child: GestureDetector(
              behavior: HitTestBehavior.opaque,
              onTap: () => Navigator.of(context).maybePop(),
              child: Padding(
                padding: EdgeInsets.symmetric(horizontal: 4, vertical: 6),
                child: Text(
                  '✕',
                  style: TextStyle(
                    color: AppColors.textMuted,
                    fontSize: 18,
                    fontWeight: AppFont.black,
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _icerik(List fikstur, int ilkGelecek, List<String> ligler) {
    if (_yukleniyor) {
      return Padding(
        padding: EdgeInsets.all(Spacing.xl),
        child: Center(
          child: CircularProgressIndicator(color: AppColors.primary),
        ),
      );
    }
    if (_hata != null) {
      // Hata GİZLENMEZ. "Maçı yok" ile "veri alınamadı" farklı şeyler.
      return Padding(
        padding: const EdgeInsets.all(Spacing.xl),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Text(
              'Fikstür alınamadı',
              style: TextStyle(
                color: AppColors.text,
                fontSize: 14,
                fontWeight: AppFont.black,
              ),
            ),
            const SizedBox(height: 6),
            Text(
              _hata!,
              textAlign: TextAlign.center,
              style: TextStyle(color: AppColors.textMuted, fontSize: 12.5),
            ),
          ],
        ),
      );
    }
    if (fikstur.isEmpty) {
      return Padding(
        padding: EdgeInsets.all(Spacing.xl),
        child: Text(
          'Bu takım için sezon maçı bulunamadı.',
          textAlign: TextAlign.center,
          style: TextStyle(color: AppColors.textMuted, fontSize: 12.5),
        ),
      );
    }

    WidgetsBinding.instance.addPostFrameCallback((_) => _ayraciGoster());

    return SingleChildScrollView(
      controller: _kaydirma,
      padding: const EdgeInsets.only(bottom: Spacing.xl),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          for (var i = 0; i < fikstur.length; i++) ...[
            if (i == ilkGelecek) _ayrac(),
            _SatirMac(f: fikstur[i] as Map, ligYaz: ligler.length > 1),
          ],
          // KAPSAM NOTU — hangi turnuvaların dahil olduğu YAZILIR.
          // Liste dolu göründüğü için, kupa/Avrupa maçlarının eksik olduğu
          // ancak burada anlaşılır. Sessiz eksik, yanlış bilgiden daha
          // tehlikelidir.
          if (ligler.isNotEmpty)
            Padding(
              padding: const EdgeInsets.only(
                left: Spacing.lg,
                right: Spacing.lg,
                top: Spacing.md,
              ),
              child: Text(
                'Kapsam: ${ligler.join(' · ')}',
                textAlign: TextAlign.center,
                style: TextStyle(
                  color: AppColors.textMuted,
                  fontSize: 10.5,
                  fontWeight: AppFont.bold,
                ),
              ),
            ),
        ],
      ),
    );
  }

  Widget _ayrac() => Padding(
    key: _ayracKey,
    padding: const EdgeInsets.symmetric(horizontal: Spacing.lg, vertical: 8),
    child: Row(
      children: [
        Expanded(child: Divider(height: 1, color: AppColors.border)),
        const SizedBox(width: 8),
        Text(
          'SIRADAKİ',
          style: TextStyle(
            color: AppColors.primary,
            fontSize: 10,
            fontWeight: AppFont.black,
            letterSpacing: 0.5,
          ),
        ),
        const SizedBox(width: 8),
        Expanded(child: Divider(height: 1, color: AppColors.border)),
      ],
    ),
  );
}

class _SatirMac extends StatelessWidget {
  const _SatirMac({required this.f, required this.ligYaz});

  final Map f;
  final bool ligYaz;

  @override
  Widget build(BuildContext context) {
    final evde = f['evde'] == true;
    final oynandi = f['oynandi'] == true;
    final sonuc = f['sonuc'] as String?;
    final skor = fiksturSkoru(f);
    final lig = '${f['lig'] ?? ''}';

    return Container(
      padding: const EdgeInsets.symmetric(vertical: 9, horizontal: Spacing.lg),
      decoration: BoxDecoration(
        color: oynandi ? null : AppColors.bgAlt,
        border: Border(bottom: BorderSide(color: AppColors.border)),
      ),
      child: Row(
        children: [
          SizedBox(
            width: 76,
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              mainAxisSize: MainAxisSize.min,
              children: [
                Text(
                  tarihEtiketi(f['dateUnix']),
                  maxLines: 1,
                  overflow: TextOverflow.ellipsis,
                  style: TextStyle(
                    color: AppColors.textMuted,
                    fontSize: 11,
                    fontWeight: AppFont.heavy,
                  ),
                ),
                // Turnuva adı YALNIZ birden fazla turnuva varsa yazılır — tek
                // turnuvada her satıra aynı adı basmak gürültüden ibaret olurdu.
                if (ligYaz && lig.isNotEmpty)
                  Padding(
                    padding: const EdgeInsets.only(top: 1),
                    child: Opacity(
                      opacity: 0.8,
                      child: Text(
                        lig,
                        maxLines: 1,
                        overflow: TextOverflow.ellipsis,
                        style: TextStyle(
                          color: AppColors.textMuted,
                          fontSize: 9,
                          fontWeight: AppFont.bold,
                        ),
                      ),
                    ),
                  ),
              ],
            ),
          ),
          Expanded(
            child: Row(
              children: [
                Expanded(
                  child: Text(
                    '${f['home'] ?? '—'}',
                    maxLines: 1,
                    overflow: TextOverflow.ellipsis,
                    style: TextStyle(
                      color: AppColors.text,
                      fontSize: 12.5,
                      // kartı açılan takım kalın
                      fontWeight: evde ? AppFont.black : AppFont.semibold,
                    ),
                  ),
                ),
                // Skor YALNIZ oynanmış maçta; oynanmamışta nötr ayraç.
                SizedBox(
                  width: 46,
                  child: Text(
                    skor ?? 'v',
                    textAlign: TextAlign.center,
                    style: TextStyle(
                      color: skor != null
                          ? AppColors.text
                          : AppColors.textMuted,
                      fontSize: skor != null ? 12.5 : 11,
                      fontWeight: skor != null ? AppFont.black : AppFont.bold,
                    ),
                  ),
                ),
                Expanded(
                  child: Text(
                    '${f['away'] ?? '—'}',
                    maxLines: 1,
                    textAlign: TextAlign.right,
                    overflow: TextOverflow.ellipsis,
                    style: TextStyle(
                      color: AppColors.text,
                      fontSize: 12.5,
                      fontWeight: evde ? AppFont.semibold : AppFont.black,
                    ),
                  ),
                ),
              ],
            ),
          ),
          const SizedBox(width: 8),
          SizedBox(
            width: 20,
            height: 20,
            child: (sonuc != null && sonuc.isNotEmpty)
                ? Container(
                    alignment: Alignment.center,
                    decoration: BoxDecoration(
                      color: _sonucRengi[sonuc] ?? AppColors.gray,
                      shape: BoxShape.circle,
                    ),
                    child: Text(
                      sonuc,
                      style: TextStyle(
                        color: okunurMetin(
                          _sonucRengi[sonuc] ?? AppColors.gray,
                        ),
                        fontSize: 10,
                        fontWeight: AppFont.black,
                      ),
                    ),
                  )
                : null,
          ),
        ],
      ),
    );
  }
}
